File: library/src/main/java/com/henorek/discharge/solutions/builders/DischargeToastBuilder.java
package com.henorek.discharge.solutions.builders;

import android.content.Context;

public class DischargeToastBuilder {

  private final Context context;

  private String message;
  private int duration;

  DischargeToastBuilder(final Context context) {
    this.context = context;
  }

  public DischargeToastBuilder withMessage(String message) {
    assert message != null;
    this.message = message;
    return this;
  }

  public DischargeToastBuilder withMessage(int messageResourceId) {
    this.message = context.getResources().getString(messageResourceId);
    return this;
  }

  public DischargeToastBuilder withDuration(int duration) {
    this.duration = duration;
    return this;
  }

  public DischargeToast build() {
    assert message != null;
    return new DischargeToast(context, message, duration);
  }
}

File: library/src/main/java/com/henorek/discharge/solutions/builders/DischargeToastBuilder.kt
package com.henorek.discharge.solutions.builders

import android.content.Context

class DischargeToastBuilder internal constructor(private val context: Context) {

  private var message: String? = null
  private var duration: Int = 0

  fun withMessage(message: String?): DischargeToastBuilder {
    assert(message != null)
    this.message = message
    return this
  }

  fun withMessage(messageResourceId: Int): DischargeToastBuilder {
    this.message = context.resources.getString(messageResourceId)
    return this
  }

  fun withDuration(duration: Int): DischargeToastBuilder {
    this.duration = duration
    return this
  }

  fun build(): DischargeToast {
    return DischargeToast(context, message!!, duration)
  }
}
